geometries/make_spe11c_geo.py

```diff
@@ -72,18 +73,18 @@ def z_offset_at(y: float) -> float:
     )
 
     print("Copying Surfaces to the back of the domain")
-    size_domain_y = 5000.0
     gmsh.model.occ.synchronize()
     frontside_surface_tags = gmsh.model.getEntities(dim=2)
     backside_surface_tags = gmsh.model.occ.copy(frontside_surface_tags)
-    gmsh.model.occ.translate(backside_surface_tags, dx=0.0, dy=size_domain_y, dz=0.0)
+    backside_dz = z_offset_at(SIZE_DOMAIN_Y)
+    gmsh.model.occ.translate(backside_surface_tags, dx=0.0, dy=SIZE_DOMAIN_Y, dz=backside_dz)
 
     def _make_connecting_spline(source_tag: int, target_tag: int) -> tuple[int, list]:
         source = gmsh.model.getValue(0, source_tag, [])
         target = gmsh.model.getValue(0, target_tag, [])
         assert abs(source[0] - target[0]) < 1e-6 \
-            and abs(source[2] - target[2]) < 1e-6 \
-            and "Expecting source and target to have only differing y coordinates"
+            and abs(abs(source[2] - target[2]) - backside_dz) < 1e-6 \
+            and "Unexpected geometric relation between front and back side points"
```

Comment:
On my machine these assertions worked, I hope they are not too strong (depending on the gmsh version and/or cpu architecture, etc)... we could also just print a warning instead of exiting?